tests/kafkatest/tests/streams/streams_standby_replica_test.py

```diff
@@ -0,0 +1,159 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ducktape.tests.test import Test
+from ducktape.utils.util import wait_until
+from kafkatest.services.kafka import KafkaService
+from kafkatest.services.streams import StreamsStandbyTaskDriverService
+from kafkatest.services.streams import StreamsStandbyTaskService
+from kafkatest.services.verifiable_consumer import VerifiableConsumer
+from kafkatest.services.zookeeper import ZookeeperService
+
+
+class StreamsStandbyTask(Test):
+    """
+    This test validates using standby tasks helps with rebalance times
+    additionally verifies standby replicas continue to work in the
+    face of continual changes to streams code base
+    """
+
+    input_topic = "standbyTaskSource1"
+    output_topic_1 = "standbyTaskSink1"
+    output_topic_2 = "standbyTaskSink2"
+
+    num_messages = 60000
+
+    def __init__(self, test_context):
+        super(StreamsStandbyTask, self).__init__(test_context=test_context)
+        self.zk = ZookeeperService(test_context, num_nodes=1)
+        self.kafka = KafkaService(test_context,
+                                  num_nodes=3,
+                                  zk=self.zk,
+                                  topics={
+                                      self.input_topic: {'partitions': 6, 'replication-factor': 1},
+                                      self.output_topic_1: {'partitions': 1, 'replication-factor': 1},
+                                      self.output_topic_2: {'partitions': 1, 'replication-factor': 1}
+                                  })
+
+    def get_consumer(self, topic, num_messages):
```

Comment:
Could we reuse the function `get_consumer` in `StreamsBrokerDownResilience`?

Reply:
Ack, for this comment and below, I was considering adding a Jira to fold several parts of the helper methods into `streams.py` for re-use across other tests.

WDYT?
\cc @dguy @mjsax

Reply:
Sounds good.

Reply:
👍